chainer/cuda.py

```diff
@@ -255,6 +256,27 @@ def to_gpu(array, device=None, stream=None):
             'Please remove it.', DeprecationWarning)
 
     check_cuda_available()
+    with Device(device) as device_:
+        if isinstance(array, (list, tuple)):
+            d = {}
+            ret = []
+            for arr in array:
+                if arr is None:
+                    ret.append(None)
+                else:
+                    arr2 = d.get(id(arr))
+                    if arr2 is None:
+                        arr2 = _array_to_gpu(arr, device_, stream)
+                        d[id(arr)] = arr2
+                    ret.append(arr2)
+            return type(array)(ret)
+        else:
+            return _array_to_gpu(array, device_, stream)
+
+
+def _array_to_gpu(array, device, stream):
+    if array is None:
+        return None
     if isinstance(array, (numpy.number, numpy.bool_)):
         array = numpy.asarray(array)
     if not isinstance(array, (cupy.ndarray, numpy.ndarray)):
```

Comment:
Shouldn't we able to send mdarray to GPU?

Reply:
Yes, we should be able to handle `mdarray`. Let's revisit after new ideep will be released.